smalltalksrc/VMMakerTests/VMPrimitiveTest.class.st

```diff
@@ -1959,6 +1959,54 @@ VMPrimitiveTest >> testPrimitiveFormatWithWeakArray [
 		equals: (memory integerObjectOf: memory weakArrayFormat)
 ]
 
+{ #category : 'tests - primitiveGetAddressOfOOPPinningIfNeeded' }
+VMPrimitiveTest >> testPrimitiveGetAddressOfOOPPinningIfNeededFailsOnImmediate [
+
+	interpreter push: (memory integerObjectOf: 2).
+	
+	interpreter primitiveGetAddressOfOOPPinningIfNeeded. 
+
+	self assert: interpreter failed.
+	
+	
+]
+
+{ #category : 'tests - primitiveGetAddressOfOOPPinningIfNeeded' }
+VMPrimitiveTest >> testPrimitiveGetAddressOfOOPPinningIfNeededReturnsAddressAndPinsIfObjectIsNotPinned [
```

Comment:
If not pinned this primitive pin it on the interpreter but not on the JIT, right?